src/index.ts

```diff
@@ -105,30 +103,45 @@ export class Tunnel extends EventEmitter {
 				.then(() =>
 					this.emit('connect', hostname, port, head)
 				);
-			}).catch((err) => {
+			}).catch((err: Error) => {
 				this.emit('error', err);
 				cltSocket.destroy();
 			})
 		);
 	}
 
+	use(middleware: Middleware) {
+		this.stack.push(middleware);
+	}
+
+	private handleMiddleware(req: Request, cltSocket: net.Socket, head: Buffer): Promise<void> {
+		return Promise.try(() => {
+			let index = 0;
+
+			const next = () => {
```

Comment:
It's possible for normal expressjs middleware to call `next` with an error argument which should skip the other middlewares, it's also possible to call it as `next('route')` which stops matching against the current route and instead sees if any others match - I'm not sure the 'route' case applies here but it probably does make sense to handle the error case

Reply:
Sure, ok. We weren't using the `Error` case so I was just aiming to keep things simple - but I'll  add it.

Reply:
It can probably just be done with either
```typescript
if (err) {
	cb(err)
	return
}
```
or
```typescript
if (err) {
	reject(err)
	return
}
```
depending on `new Promise` or `Promise.fromCallback`